jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/TransitNodeRoutingPrecomputation.java

```diff
@@ -0,0 +1,1214 @@
+
+            if (sourceVisitedVoronoiCells.contains(NO_VORONOI_CELL)
+                    || sinkVisitedVoronoiCells.contains(NO_VORONOI_CELL)) {
+                return true;
+            }
+
+            Set<Integer> smallerSet;
+            Set<Integer> largerSet;
+            if (sourceVisitedVoronoiCells.size() <= sinkVisitedVoronoiCells.size()) {
+                smallerSet = sourceVisitedVoronoiCells;
+                largerSet = sinkVisitedVoronoiCells;
+            } else {
+                smallerSet = sinkVisitedVoronoiCells;
+                largerSet = sourceVisitedVoronoiCells;
+            }
+
+            for (Integer visitedVoronoiCell : smallerSet) {
+                if (largerSet.contains(visitedVoronoiCell)) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    }
+
+    /**
+     * Stores forward and backward access vertices computed for the
+     * transit node routing.
+     *
+     * @param <V> graph vertex type
+     * @param <E> graph edge type
+     */
+    public static class AccessVertices<V, E> {
+        /**
+         * For each vertex in {@code contractionGraph} stores corresponding forward access vertices.
+         */
+        private List<List<AccessVertex<V, E>>> forwardAccessVertices;
+        /**
+         * For each vertex in {@code contractionGraph} stores corresponding backward access vertices.
+         */
+        private List<List<AccessVertex<V, E>>> backwardAccessVertices;
+
+        /**
+         * Constructs a new instance for the given {@code forwardAccessVertices} and {@code backwardAccessVertices}.
+         *
+         * @param forwardAccessVertices  forward access vertices
+         * @param backwardAccessVertices backward access vertices
+         */
+        public AccessVertices(List<List<AccessVertex<V, E>>> forwardAccessVertices,
+                              List<List<AccessVertex<V, E>>> backwardAccessVertices) {
+            this.forwardAccessVertices = forwardAccessVertices;
+            this.backwardAccessVertices = backwardAccessVertices;
+        }
+
+        /**
+         * Given a contraction vertex {@code vertex} returns its forward access vertices
+         *
+         * @param vertex vertex
+         * @return list of forward access vertices
+         */
+        public List<AccessVertex<V, E>> getForwardAccessVertices(ContractionVertex<V> vertex) {
+            return forwardAccessVertices.get(vertex.vertexId);
+        }
+
+        /**
+         * Given a contraction vertex {@code vertex} returns its backward access vertices
+         *
+         * @param vertex vertex
+         * @return list of backward access vertices
+         */
+        public List<AccessVertex<V, E>> getBackwardAccessVertices(ContractionVertex<V> vertex) {
+            return backwardAccessVertices.get(vertex.vertexId);
+        }
+    }
+
+    /**
+     * Forward or backward access vertex computed for a certain vertex $v$ in the graph.
```

Comment:
This is a public class and its Javadoc will be visible to users. Please, add a brief explanation of how this vertex is used and what it's needed for. Most users don't have a relative background, so the current Javadoc won't be clear to them.